[WIP] OSDOCS#12884: Manually create RBAC to install an extension (#87627)
openshift/openshift-docs: modules/olmv1-cluster-extension-permissions.adoc

// Module included in the following assemblies:
//
// * extensions/ce/managing-ce.adoc

:_mod-docs-content-type: CONCEPT

[id="olmv1-cluster-extension-permissions_{context}"]
= Cluster extension permissions

> {olmv1-first} is designed to be secure by default. Unlike {olmv0}, {olmv1} does not have cluster administrator permissions. As a result, cluster administrators must create a service account for each cluster extension and assign the role-based access controls (RBAC) required to install and manage that extension.

Reviewer: wdyt if we rephrase it to be something like:

```suggestion
{olmv1-first} is designed to be secure by default. Rather than managing all cluster extensions with a single cluster administrator scoped service account, as in {olmv0}, {olmv1} uses manages each cluster extension as the service account specified in its CR. As a result, cluster administrators can create a service account for each cluster extension and assign role-based access controls (RBAC) with only the permissions required to install and manage that extension.
```
(cc @joelanford)

Author: I took a stab at incorporating your feedback here: https://github.com/openshift/openshift-docs/pull/87627/commits/fae63c316ff38806258ab3a372ab41e101ca35b8#diff-6bd37e1983027f833fe453f5082e41498125ca7d4e045760fd40575eee8b84ddR10

WDYT?

> You must add each permission to either a cluster role or role. Then you must bind the permission to the service account with a cluster role binding or role binding. Use cluster roles and cluster role bindings to scope permissions to the cluster. Use roles and role bindings to scope permissions to a namespace.

Reviewer: ```suggestion
You must add each permission to either a cluster role or role. Then you must bind the cluster role or role to the service account with a cluster role binding or role binding. Use cluster roles and cluster role bindings to scope permissions to the cluster. Use roles and role bindings to scope permissions to a namespace.
```

Author: Addressed: https://github.com/openshift/openshift-docs/pull/87627/commits/fae63c316ff38806258ab3a372ab41e101ca35b8#diff-6bd37e1983027f833fe453f5082e41498125ca7d4e045760fd40575eee8b84ddR12

// Module included in the following assemblies:
//
// * extensions/ce/managing-ce.adoc

:_mod-docs-content-type: CONCEPT

[id="olmv1-cluster-extension-permissions_{context}"]
= Cluster extension permissions

In {olmv0-first}, a single service account with cluster administrator privileges manages all cluster extensions.

{olmv1} is designed to be secure by default. {olmv1} manages a cluster extension by using the service account specified in an extension's custom resource (CR). Cluster administrators can create a service account for each cluster extension. As a result, you can follow the principle of least privilege and assign only the role-based access controls (RBAC) to install and manage that extension.

> You must add each permission to either a cluster role or role. Then you must bind the cluster role or role to the service account with a cluster role binding or role binding. Use cluster roles and cluster role bindings to scope permissions to the cluster. Use roles and role bindings to scope permissions to a namespace.

Author: The rub is that these are not CRs, but specific Kubernetes `ClusterRole` and `Role` objects. So, if I were to use parallelism, I would need to write "cluster-scoped cluster roles" and "namespace-scoped roles". 

I took another approach, to address your feedback. Hopefully, that clarifies things a bit. 

I have follow up work after GA on this section, so there is also a chance that additional context and information in the future will help here.

// Module included in the following assemblies:
//
// * extensions/ce/managing-ce.adoc

:_mod-docs-content-type: CONCEPT

[id="olmv1-cluster-extension-permissions_{context}"]
= Cluster extension permissions

In {olmv0-first}, a single service account with cluster administrator privileges manages all cluster extensions.

{olmv1} is designed to be more secure than {olmv0} by default. {olmv1} manages a cluster extension by using the service account specified in an extension's custom resource (CR). Cluster administrators can create a service account for each cluster extension. As a result, administrators can follow the principle of least privilege and assign only the role-based access controls (RBAC) to install and manage that extension.

You must add each permission to either a cluster role or role. Then you must bind the cluster role or role to the service account with a cluster role binding or role binding.

> You can scope the RBAC to either the cluster or to a namespace. Use cluster roles and cluster role bindings to scope permissions to the cluster. Use roles and role bindings to scope permissions to a namespace. Whether you scope the permissions to the cluster or to a namespace depends the design of the extension you want to install and manage.

Reviewer 2: ```suggestion
You can scope the RBAC to either the cluster or to a namespace. Use cluster roles and cluster role bindings to scope permissions to the cluster. Use roles and role bindings to scope permissions to a namespace. Whether you scope the permissions to the cluster or to a namespace depends on the design of the extension you want to install and manage.
```

Author: addressed: https://github.com/openshift/openshift-docs/pull/87627/commits/7cee26fbeaf56af6ea948a3862d3e31af60ffeea
